### ISSUE TITLE
Feature request: separate file-related installation and DB-related installation for easier Docker use

### ISSUE THREAD

**Issue by KingMob** [netz98/n98-magerun#554]
First, thanks for such a great tool!

It would be nice to be able to run the installation of files in the Magento root dir separately from installation to the database, so that people using Docker can maintain separate containers for each. 

As it stands, n98-magerun's install updates both, so either the installation occurs after the Docker containers are running, in which case you have to `docker commit` the images in order to save your changes (not ideal), or you have to maintain a single container combining both, which makes it harder to maintain multiple versions of Magento for testing (since older versions have different MySQL versions allowed).

I don't know Magerun well enough to make PR yet, but perhaps it's as simple as making the call to `createDatabase` optional, as well as breaking it out into a separate command of its own?

**Comment by tkn98** [comment]
Yes, the idea is a good one. There are even flags internally, from what I've seen last time it was not that easy.

So to say: This requires some reading of the sources and testing. So if you're interested feel free to dig a bit with it. I think there is a benefit in making those two parts of the installation more modular.

Next to file and database installation, there is also one command to generate the local.xml file. It is a key-file to signal Magento that an installation was done.

**Comment by KingMob** [comment]
Maybe when I have more time. Sadly, I'm working 7 days a week in my new position. 

Could the presence of the local.xml just be a lie until the DB part is done? Unfortunately, the DB connection details wouldn't be available until the containers are linked and running, so they can't be done in the Dockerfile, without mandating the connection info in advance (which will still interfere with running multiple Magento versions.) And the DB part can't write the file since it won't know beforehand the values of its conn params to the PHP container.

I'm not sure how to proceed, other than by overriding how Magento loads config info to use env variables.

I suppose we could make the startup docker CMD be something like:
generate-magento-local-xml.sh && apache2 -D FOREGROUND

This could work as long as people tolerate the non-existence of the local.xml file when doing anything else with the container/image.
